server/src/test/java/org/opensearch/tasks/TaskCancellationMonitoringServiceTests.java

```diff
@@ -0,0 +1,321 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.tasks;
+
+import org.junit.After;
+import org.junit.Before;
+import org.opensearch.Version;
+import org.opensearch.action.search.SearchShardTask;
+import org.opensearch.common.io.stream.StreamOutput;
+import org.opensearch.common.settings.ClusterSettings;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.test.OpenSearchTestCase;
+import org.opensearch.test.transport.MockTransportService;
+import org.opensearch.threadpool.Scheduler;
+import org.opensearch.threadpool.TestThreadPool;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.transport.TransportRequest;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Phaser;
+import java.util.concurrent.TimeUnit;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+import static org.opensearch.tasks.TaskCancellationMonitoringSettings.DURATION_MILLIS_SETTING;
+
+public class TaskCancellationMonitoringServiceTests extends OpenSearchTestCase {
```

Comment:
Can we add unit test for processing large number of tasks (100k-1m) to ensure that the processing of MonitoringService is O(n) and any regression can be caught easily.

Reply:
I am not sure about adding these many tasks(100k) to UT's which involves creating, cancelling these. While checking existing UTs I don't see such a large number being used. Will wait for other's opinion on this. @Bukhtawar @andrross @dblock

Reply:
Created an issue for this - https://github.com/opensearch-project/OpenSearch/issues/7976